src/gui/tray/Window.qml

```diff
@@ -720,43 +720,45 @@ Window {
                             anchors.right: moreActionsButton.right
                             anchors.top: moreActionsButton.top
 
-                            // transform model to contain indexed actions with primary action filtered out
-                            function actionListToContextMenuList(actionList, maxButtons) {
-                                // early exit when we have less than maxButtons actions
-                                if (actionList.length <= maxButtons) {
-                                    return actionList;
-                                }
-
-                                // add index to every action
-                                var actionsWithIndices = actionList.map(function(action, index) {
+                            Menu {
+                                id: moreActionsButtonContextMenu
+                                anchors.centerIn: parent
 
-                                    var actionWithIndex = { actionIndex: index };
+                                // transform model to contain indexed actions with primary action filtered out
+                                function actionListToContextMenuList(actionList, skipTransform) {
+                                    // early out with non-altered data
+                                    if (skipTransform === true) {
+                                        return actionList;
+                                    }
 
-                                    Object.assign(actionWithIndex, action)
+                                    // add index to every action
+                                    var actionsWithIndices = actionList.map(function(action, index) {
 
-                                    return actionWithIndex;
-                                });
+                                        var actionWithIndex = { actionIndex: index, label: action.label, primary: action.primary };
 
-                                // filter 'primary' action out
-                                var filteredActions = actionsWithIndices.filter(function(action) {
-                                    return action.primary !== true;
-                                });
+                                        return actionWithIndex;
+                                    });
 
-                                return filteredActions;
-                            }
+                                    // filter 'primary' action out
+                                    var filteredActions = actionsWithIndices.filter(function(action) {

```

Comment:
Just wondering, wouldn't it be better to filter first and map second? Is that possible or the nature of the elements in the input list get in the way?

Reply:
@er-vin Not really. I need an index of the element in the list to use it in 'actionIndex'. Otherwise, if I filter elements first, then I will not have the information to use for 'actionIndex' as the order and the number of the elements would change :(

Actually, I've got an idea. Let me give it a try...

Reply:
> @er-vin Not really. I need an index of the element in the list to use it in 'actionIndex'. Otherwise, if I filter elements first, then I will not have the information to use for 'actionIndex' as the order and the number of the elements would change :(

Oh right...

> Actually, I've got an idea. Let me give it a try...

Well, only if it's easy, otherwise I'd say: don't bother.

Reply:
@er-vin Actually it happened to be quite easy. Check my last commit with the final improvement.